src/FSharpPlus/Extensions/List.fs

```diff
@@ -146,3 +146,12 @@ module List =
             | (l::ls,r::rs) -> loop ((l,r)::acc) (ls,rs)
             | (_,_) -> acc
         loop [] (l1,l2) |> List.rev
+        
+    /// <summary>Applies given function to each value of the given List.</summary>
+    /// <param name="f">The mapping function.</param>
+    /// <param name="x">The input list.</param>
+    ///
+    /// <returns>List with values x for each List value where the function returns Some(x).</returns>
+    let chooseIndex f l =
+        List.indexed l
```

Comment:
There might be a more efficient implementation that traverses the list only once. Maybe having a look at the `choose` implementation in FSharp.Core and adding the index is a good idea.

Reply:
Changed implementation to a recursive one